activiti-engine/src/main/java/org/activiti/engine/impl/persistence/entity/integration/IntegrationContextEntityImpl.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2017 Alfresco, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.activiti.engine.impl.persistence.entity.integration;
+
+import java.util.Date;
+import java.util.HashMap;
+
+import org.activiti.engine.impl.persistence.entity.AbstractEntity;
+
+public class IntegrationContextEntityImpl extends AbstractEntity implements IntegrationContextEntity {
+
+    private String executionId;
+
+    private String correlationId;
+
+    private Date createdDate;
+
+    private String processInstanceId;
+
+    private String processDefinitionId;
+
+    @Override
+    public String getExecutionId() {
+        return executionId;
+    }
+
+    @Override
+    public void setExecutionId(String executionId) {
+        this.executionId = executionId;
+    }
+
+    @Override
+    public String getCorrelationId() {
+        return correlationId;
+    }
+
+    @Override
+    public void setCorrelationId(String correlationId) {
+        this.correlationId = correlationId;
+    }
+
+    public String getProcessDefinitionId() {
+        return processDefinitionId;
+    }
+
+    public void setProcessDefinitionId(String processDefinitionId) {
+        this.processDefinitionId = processDefinitionId;
+    }
+
+    public String getProcessInstanceId() {
+        return processInstanceId;
+    }
+
+    public void setProcessInstanceId(String processInstanceId) {
+        this.processInstanceId = processInstanceId;
+    }
+
+    public Date getCreatedDate() {
+        return createdDate;
+    }
+
+    public void setCreatedDate(Date createdDate) {
+        this.createdDate = createdDate;
+    }
+
+    @Override
+    public Object getPersistentState() {
```

Comment:
This is used for internal cache purposes. It's here because we are relying on the core engine persistence mechanism that requires this information. This does not refer to the data that is send out. See my previous comment.

Reply:
@erdemedeiros understood.. so this is a requirement for all entities?

Reply:
Exactly